### ISSUE TITLE
[Bug]: JSON with unescaped quotes gets incorrectly parsed

### ISSUE THREAD

**Issue by sebablanco** [mangiucugna/json_repair#85]
### Version of the library

0.30.3

### Describe the bug

Since 0.29.2, if a JSON object contains a string that has unescaped quotes, json_repair seems to be parsing it incorrectly. 

We had one LLM produce unescaped quotes so we started using `json_repair` and added test cases for it. Up to version 0.29.1, json_repair would be able to fix it as expected. In the versions since it doesn't.

### How to reproduce

This short sample script will succeed on 0.29.1, but will print a diff in versions since:
```python
import difflib

import json_repair

test = """
{"type": "criteria_score", "score": 3, "comment": "The essay uses some technical vocabulary related to soccer, such as "offside" and "penalty". To improve, try to incorporate more precise, subject-specific language to enhance the technical accuracy."}
"""
expected = {
    "type": "criteria_score",
    "score": 3,
    "comment": 'The essay uses some technical vocabulary related to soccer, such as "offside" and "penalty". To improve, try to incorporate more precise, subject-specific language to enhance the technical accuracy.',
}


loaded_json = json_repair.loads(test)
if loaded_json != expected:
    diff = difflib.ndiff(str(loaded_json).splitlines(), str(expected).splitlines())
    print("\n".join(diff))
else:
    print("The loaded JSON matches the expected dictionary.")
```

### Expected behavior

We'd expect the parsed json to match the `expected` dict in the code above.

**Comment by mangiucugna** [comment]
Apologies for the regression, I have updated tests and about to release 0.31 with the fix. Let me know if that fixes the issue as the tests are passing.

**Comment by zmedelis** [comment]
Testing with 0.31.0

```
txt = '["good item", "bad "item""]'
>>> json_repair.loads(txt)
['good item', 'bad', 'item']
````

In my case I was expecting `['good item', 'bad "item"']`. Are there any cases where `['good item', 'bad', 'item']` might be the expected result?

**Comment by mangiucugna** [comment]
Hi apologies for the late reply but I got busy with work. I have released 0.33.0 with what would should be a better handling of this corner case for arrays (since it worked for objects)
